components/brave_wallet/browser/zcash/zcash_wallet_service.cc

```diff
@@ -375,6 +376,7 @@ void ZCashWalletService::OnCompleteTransactionTaskDone(
     SignAndPostTransactionCallback callback,
     base::expected<ZCashTransaction, std::string> result) {
   CHECK(complete_transaction_tasks_.erase(task));
+  CHECK(ValidateTransaction(original_zcash_transaction));
```

Comment:
IMO we should not crash here, but just fail with InternalError message.
If we want to see this occurring for users we should sent a historgram(which we can't) or do `DumpWithoutCrashing`

Reply:
https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/checks.md#invariant_verification-mechanisms